core/federated/RTI/rti_common.h

```diff
@@ -49,63 +50,67 @@ typedef struct enclave_t {
-    lf_cond_t next_event_condition; // Condition variable used by enclaves to notify an enclave
-                                    // that it's call to next_event_tag() should unblock.
-} enclave_t;
+} reactor_node_info_t;
 
 /**
- * Structure that an enclave RTI instance uses to keep track of its own and its
- * corresponding enclaves'state.
- *     // **************** IMPORTANT!!! ********************
- *     // **   If you make any change to this struct,     **
- *     // **  you MUST also change federation_rti_t in    **
- *     // ** (rti_lib.h)! The change must exactly match.  **
- *     // **************************************************
+ * Data structure which is common to both the remote standalone RTI and the local RTI used in enclaved execution.
+ * rti_remote_t and rti_local_t will "inherit" from this data structure. The first field is an array of pointers 
+ * to reactor_nodes. These will be reactor_nodes for the local RTI and federates for the remote RTI 
+ * 
  */
+typedef struct rti_common_t {
+    // The reactor nodes.
+    reactor_node_info_t **reactor_nodes;
 
-typedef struct enclave_rti_t {
-    // The enclaves.
-    enclave_t **enclaves;
+    // Number of reactor nodes
+    int32_t number_of_reactor_nodes;
 
-    // Number of enclaves
-    int32_t number_of_enclaves;
-
-    // RTI's decided stop tag for enclaves
+    // RTI's decided stop tag for the reactor nodes
     tag_t max_stop_tag;
 
-    // Number of enclaves handling stop
-    int num_enclaves_handling_stop;
+    // Number of reactor nodes handling stop
+    int num_reactor_nodes_handling_stop;
 
     // Boolean indicating that tracing is enabled.
     bool tracing_enabled;
-} enclave_rti_t;
 
+    // Lock for making thread-safe access to the shared state
+    lf_mutex_t mutex;
+} rti_common_t;
+
+typedef struct {
+    tag_t tag;           // NEVER if there is no tag advance grant.
+    bool is_provisional; // True for PTAG, false for TAG.
+} tag_advance_grant_t;
+
+/**
+ * @brief Initialize the fields of the rti_common struct. It also stores
+ * the pointer to the struct and uses it internally.
+ * 
+ * @param rti_common 
+ */
+void initialize_rti_common(rti_common_t * rti_common);
 
 /**
  * An enclave calls this function after it completed a tag. 
- * The function updates the completed tag and check if the downstream enclaves 
+ * The function updates the completed tag and check if the downstream reactor_nodes 
  * are eligible for receiving TAGs.
  * 
  * @param enclave The enclave
  * @param completed The completed tag of the enclave
  */
-void logical_tag_complete(enclave_t* enclave, tag_t completed);
-
-typedef struct {
-    tag_t tag;           // NEVER if there is no tag advance grant.
-    bool is_provisional; // True for PTAG, false for TAG.
-} tag_advance_grant_t;
+void logical_tag_complete(reactor_node_info_t* enclave, tag_t completed);
 
 /** 
- * Initialize the enclave with the specified ID.
+ * Initialize the reactor- with the specified ID.
```

Comment:
Here, `enclave` seems like the right term. The `@param` entries are now inconsistent with the doc.